pykafka/cluster.py

```diff
@@ -446,5 +446,7 @@ def update(self):
             except LeaderNotAvailable:
                 log.warning("LeaderNotAvailable encountered. This may be "
                             "because one or more partitions have no available replicas.")
+                if i == self._max_connection_retries - 1:
+                    raise LeaderNotAvailable
```

Comment:
This could be changed to simply `raise` since we're raising an exception of the same type we're catching.